src/doc/tock_binary_format.md

```diff
@@ -586,6 +587,37 @@ Object has both, the kernel decides which to use. Tock is transitioning to
 having the Program Header as the standard one to use, but older kernels (2.0 and
 earlier) do not recognize it and use the Main Header.
 
+#### `10` ShortID
+
+```rust
+struct TbfHeaderV2ShortId {
+    base: TbfHeaderTlv,
+    short_id: u32,
+}
+```
+
+This header allows the compile-time workflow to specify the fixed `ShortId` the
+application should be assigned when it is loaded. The header only includes
+the 32 bit ShortId.
+
+Note, fixed `ShortId`s are defined to be nonzero. Therefore, including this
+header but setting the `short_id` field to 0 is the same as not including this
+header.
```

Comment:
Should this be an error instead of just silently ignoring it?

Reply:
An error where?

Reply:
I think Alistair is suggesting the kernel should throw an error when loading this process instead of ignoring the header.

I see his point, but don't feel strongly either way about what should be done. Perhaps we can just use type system here and equivalents in supporting tools to make this a largely impossible case, such that the behavior in the broken corner case is less significant.

Reply:
I really don't think we want to brick an app because of an optional TBF header.

In any case, this documents the format, and not the process loading implementation.

Reply:
Exactly. If a user is trying to load an invalid header we should throw and error so they know their process isn't loaded. Instead of just ignoring the data they provided.

Reply:
> In any case, this documents the format, and not the process loading implementation.

You are documenting the action as well when you say "field to 0 is the same as not including this header." Setting a field to 0 is an error as 0 is reserved for the kernel.

Reply:
Ok fair, I can remove that line. But the header parser won't return an error.